File: 0001_easy_hot_two-sum/solution.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
    d := make(map[int]int)
    
    for i, num := range nums {
        complement := target - num
        if val, ok := d[complement]; ok && val != i {
            return []int{i, val}
        }
        d[num] = i
    }
    
    return nil
}

func main() {
    nums := []int{2, 7, 11, 15}
    target := 9
    result := twoSum(nums, target)
    fmt.Println(result) // 输出: [1, 0]
}

File: 0001_easy_hot_two-sum/solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
                Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            
            if (map.containsKey(complement) && map.get(complement) != i) {
                return new int[]{map.get(complement), i};
            }
            
            map.put(nums[i], i);
        }
        
        // 如果没有找到符合条件的索引对，这里可以返回一个空数组或者抛出异常
        return new int[]{};
    }
}